pkg/registration/spoke/bootstrapkubeconfigs.go

```diff
@@ -0,0 +1,75 @@
+package spoke
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	clusterv1client "open-cluster-management.io/api/client/cluster/clientset/versioned"
+
+	commonhelpers "open-cluster-management.io/ocm/pkg/common/helpers"
+)
+
+func selectBootstrapKubeConfigs(ctx context.Context, managedCluster string, bootstrapKubeConfigs []*rest.Config) (int, error) {
+	logger := log.FromContext(ctx)
+
+	for index, kc := range bootstrapKubeConfigs {
+		// Send sarr check if bootstrapkubeconfig has right permission
+		bootstrapKubeClient, err := kubernetes.NewForConfig(kc)
+		if err != nil {
+			logger.Error(err, "Failed to create bootstrap kube client", "index", index)
+			continue
+		}
+		allowed, failedReview, err := commonhelpers.CreateSelfSubjectAccessReviews(ctx, bootstrapKubeClient, commonhelpers.GetBootstrapSSARs())
+		if err != nil {
+			logger.Error(err, "Failed to create self subject access reviews", "index", index)
+			continue
+		}
+		if !allowed {
+			logger.Error(err, "Failed to create self subject access reviews", "index", index, "failedReview", failedReview)
+			continue
+		}
+
+		// Get managed cluster, if managed cluster exist, then check the hubAcceptClient
+		bootstrapClusterClient, err := clusterv1client.NewForConfig(kc)
+		if err != nil {
+			logger.Error(err, "Failed to create bootstrap cluster client", "index", index)
+			continue
+		}
+
+		mc, err := bootstrapClusterClient.ClusterV1().ManagedClusters().Get(ctx, managedCluster, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return index, nil
+		}
+		if err != nil {
+			logger.Error(err, "Failed to get managed cluster", "index", index)
+			continue
+		}
+		if mc.Spec.HubAcceptsClient {
+			return index, nil
+		}
+	}
+
+	return -1, fmt.Errorf("no bootstrap kubeconfig is available for managed cluster %s", managedCluster)
+}
+
+type reSelectChecker struct {
```

Comment:
comments is needed. What is for? and how shouldReSelect should be set.

Reply:
Added.